.github/workflows/build-and-test.yml

```diff
@@ -50,13 +50,24 @@ jobs:
     - name: Setup the workspace
       run: ./gradlew --info --stacktrace ${{ inputs.workspace }}
 
-    - name: Build the mod
+    - name: Compile the mod
+      run: ./gradlew --info --stacktrace assemble jar reobfJar
+
+    - name: Attach compilation artifacts
+      uses: actions/upload-artifact@v3
+      with:
+        name: build-libs
+        path: build/libs/
+        retention-days: 31
+
+    - name: Run post-build checks
```

Comment:
It's a bit confusing to label build as post-build checks.. check maybe?

Reply:
The issue is forgegradle only runs some necessary things in the build process via the `build` task, so `check` is less reliable